libs/apigateway-lambda-proxy/src/lib/metric-alarms.ts

```diff
@@ -0,0 +1,376 @@
+             * each resource
+             */
+            getTags: (
+                name: string,
+            ) => {
+                [key: string]: pulumi.Input<string>
+            }
+        },
+        opts?: pulumi.ComponentResourceOptions,
+    ) {
+        super('wanews:apigateway/ErrorRate4xxAlarm', name, {}, opts)
+
+        const metricName = isHttpApi(args.apiGateway) ? '4xx' : '4XXError'
+        const dimensions = getDimensions(args.apiGateway)
+        const thresholdPercent = args.errorRate4xxPercent ?? 50
+        const threshold = pulumi
+            .output(thresholdPercent)
+            .apply((percent) => percent / 100)
+        const resourceName = `${name}-4xx-rate`
+
+        new aws.cloudwatch.MetricAlarm(
+            resourceName,
+            {
+                evaluationPeriods: 2,
+                datapointsToAlarm: 2,
+                comparisonOperator: 'GreaterThanOrEqualToThreshold',
+                namespace,
+                metricName,
+                dimensions,
+                threshold,
+                statistic: 'Average',
+                alarmDescription: pulumi.interpolate`threshold: 4xx rate >= ${thresholdPercent}%: ${
+                    args.apiGateway.name ?? args.apiGateway.id
+                }`,
+                alarmActions: [args.snsTopicArn],
+                okActions: [args.snsTopicArn],
+                insufficientDataActions: [args.snsTopicArn],
+                treatMissingData: 'notBreaching',
+                tags: args.getTags(resourceName),
+            },
+            {
+                parent: this,
+                deleteBeforeReplace: true,
+            },
+        )
+    }
+}
+
+/** raise an alarm if integration latency is too high */
+export class IntegrationLatencyAlarm extends pulumi.ComponentResource {
+    constructor(
+        name: string,
+        args: {
+            /**
+             * Alerts will be sent to this SNS topic
+             */
+            snsTopicArn: pulumi.Input<string>
+            /**
+             * The API Gateway to monitor
+             */
+            apiGateway: ApiGateway
+            /**
+             * The number of standard deviations to use for the anomaly detection.
+             *
+             * Default: 5
+             */
+            stdDeviations?: pulumi.Input<number>
+            /**
+             * a callback function that returns tags for
+             * each resource
+             */
+            getTags: (
+                name: string,
+            ) => {
+                [key: string]: pulumi.Input<string>
+            }
+        },
+        opts?: pulumi.ComponentResourceOptions,
+    ) {
+        super('wanews:apigateway/IntegrationLatencyAlarm', name, {}, opts)
+
+        const resourceName = `${name}-integration-latency`
+        const metricName = 'IntegrationLatency'
+        const dimensions = getDimensions(args.apiGateway)
+        args.stdDeviations = args.stdDeviations ?? 5
```

Comment:
Why 5 std deviations out of interest?

Reply:
This is just a default, but it picks up the recent incidents in the-game.